Arrangement_on_surface_2/include/CGAL/Arr_overlay_2.h

```diff
@@ -40,6 +40,81 @@
 
 namespace CGAL {
 
+template <typename Arr1, typename Arr2, typename Curve>
+class Indexed_sweep_accessor
+{
+  const Arr1& arr1;
+  const Arr2& arr2;
+  mutable std::vector<void*> backup_inc;
+
+public:
+
+  Indexed_sweep_accessor (const Arr1& arr1, const Arr2& arr2)
+    : arr1(arr1), arr2(arr2) { }
+
+  std::size_t nb_vertices() const
+  {
+    return arr1.number_of_vertices() + arr2.number_of_vertices();
+  }
+
+  std::size_t min_end_index (const Curve& c) const
+  {
+    return reinterpret_cast<std::size_t>(halfedge(c)->target()->inc());
+  }
+
+  std::size_t max_end_index (const Curve& c) const
+  {
+    return reinterpret_cast<std::size_t>(halfedge(c)->source()->inc());
+  }
+
+  const Curve& curve (const Curve& c) const
+  {
+    return c;
+  }
+
+  void before_init() const
+  {
+    std::size_t idx = 0;
+    backup_inc.reserve (nb_vertices());
+    for (typename Arr1::Vertex_const_iterator vit = arr1.vertices_begin();
+         vit != arr1.vertices_end(); ++vit, ++idx)
+    {
+      backup_inc[idx] = vit->inc();
+      vit->set_inc (reinterpret_cast<void*>(idx));
+    }
+    for (typename Arr2::Vertex_const_iterator vit = arr2.vertices_begin();
+         vit != arr2.vertices_end(); ++vit, ++idx)
+    {
+      backup_inc[idx] = vit->inc();
+      vit->set_inc (reinterpret_cast<void*>(idx));
+    }
+  }
+
+  void after_init() const
```

Comment:
add a comment saying that you are restoring the property after the squat

Reply:
@sloriot Is this blocking the merge?

Reply:
I think the comment is here now. Marking it as resolved.